docs/migration-0.3-0.4.md

```diff
@@ -0,0 +1,42 @@
+# Topiary CLI Migration Guide
+Topiary 0.4 has switched away from toml to [Nickel](https://nickel-lang.org/) for its configuration language.
+For a help on how to configure Topiary using Nickel see the [`README`](/README.md).
+This guide is only there to help you migrate from toml to Nickel.
+
+### Migrating to Nickel
+## Before
+```toml
+[[language]]
+name = "bash"
+extensions = ["sh", "bash"]
+
+[[language]]
+name = "rust"
+extensions = ["rs"]
+indent = "    " # 4 spaces
+```
+
+## After
+```nickel
+{
+  "bash" = {
+    extensions = ["sh", "bash"],
+  },
+
+  "rust" = {
+    extensions = ["rs"],
+    indent = "    ", # 4 spaces
+  },
+}
```

Comment:
In the `README` and `topiary-config/languages.ncl`, you do:

```nickel
{
  languages = {
    <LANGUAGE> = {
      extensions = ["foo"],
    }
  }
}
```

i.e., There's a top-level `languages` key. Is it needed here?